### ISSUE TITLE
Seeding database fails

### ISSUE THREAD

**Issue by ojongerius** [freeCodeCamp/freeCodeCamp#17257]
If I try to seed a database it fails. How to reproduce:

* `docker-compose down` # make sure none of the related containers are running
* `docker system prune -a` # remove all stopped containers, images and networks not in use
* `docker-compose run --rm freecodecamp npm install`#  install deps
* `docker-compose run --rm freecodecamp npm run only-once` # seed

Of interest is that ` entities not found, db may not be properly seeded` is ironic when you try to seed a db 😉 

Related but not identical #17256 -which shows a timeout connecting to the db, while my setup connects to the db just fine (`fcc:server db connected +0ms`).

```sh
▶ docker-compose run --rm freecodecamp npm run only-once
Starting freecodecamp_mailhog_1 ... done
Starting freecodecamp_db_1      ... done

> freecodecamp@0.1.0 only-once /app
> npm run prelint-js && echo '/****/' && echo 'Seeding Database' && echo '/****/' && node seed/index.js && echo '/****/' && echo 'Seeding Completed' && echo '/****/'


> freecodecamp@0.1.0 prelint-js /app
> npm run ensure-env


> freecodecamp@0.1.0 ensure-env /app
> node ./config/ensure-env.js



rev-manifest present




pathMigration present


/****/
Seeding Database
/****/
Warning: Accessing createClass via the main React package is deprecated, and will be removed in React v16.0. Use a plain JavaScript class instead. If you're not yet ready to migrate, create-react-class v15.* is available on npm as a temporary, drop-in replacement. For more info see https://fb.me/react-create-class
Warning: Accessing PropTypes via the main React package is deprecated, and will be removed in  React v16.0. Use the latest available v15.* prop-types package from npm instead. For info on usage, compatibility, migration and more, see https://fb.me/prop-types-docs
  fcc:server db connected +0ms
/app/node_modules/loopback-connector-mongodb/node_modules/mongodb/lib/utils.js:123
    process.nextTick(function() { throw err; });
                                  ^

Error: Error: entities not found, db may not be properly seeded
    at AnonymousObserver._onError (/app/server/boot/home.js:18:21)
    at AnonymousObserver.Rx.AnonymousObserver.AnonymousObserver.error (/app/node_modules/rx/dist/rx.js:1836:12)
    at AnonymousObserver.Rx.internals.AbstractObserver.AbstractObserver.onError (/app/node_modules/rx/dist/rx.js:1772:14)
    at AnonymousObserver.o.onError (/app/node_modules/rx/dist/rx.js:1999:22)
    at AnonymousObserver.tryCatcher (/app/node_modules/rx/dist/rx.js:63:31)
    at AutoDetachObserverPrototype.error (/app/node_modules/rx/dist/rx.js:5891:52)
    at AutoDetachObserver.Rx.internals.AbstractObserver.AbstractObserver.onError (/app/node_modules/rx/dist/rx.js:1772:14)
    at InnerObserver.error (/app/node_modules/rx/dist/rx.js:4530:14)
    at InnerObserver.Rx.internals.AbstractObserver.AbstractObserver.onError (/app/node_modules/rx/dist/rx.js:1772:14)
    at InnerObserver.o.onError (/app/node_modules/rx/dist/rx.js:1999:22)
    at InnerObserver.tryCatcher (/app/node_modules/rx/dist/rx.js:63:31)
    at AutoDetachObserverPrototype.error (/app/node_modules/rx/dist/rx.js:5891:52)
    at AutoDetachObserver.Rx.internals.AbstractObserver.AbstractObserver.onError (/app/node_modules/rx/dist/rx.js:1772:14)
    at AutoDetachObserver.o.onError (/app/node_modules/rx/dist/rx.js:1999:22)
    at AutoDetachObserver.o.onError (/app/node_modules/rx/dist/rx.js:1999:22)
    at /app/node_modules/rx/dist/rx.js:1904:72
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! freecodecamp@0.1.0 only-once: `npm run prelint-js && echo '/****/' && echo 'Seeding Database' && echo '/****/' && node seed/index.js && echo '/****/' && echo 'Seeding Completed' && echo '/****/'`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the freecodecamp@0.1.0 only-once script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2018-05-29T00_05_10_316Z-debug.log
```

**Comment by ojongerius** [comment]
@rajatkantinandi ^^

**Comment by ptsurbeleu** [comment]
I looked into the runtime state and it looks like #17150 could be the reason why seeding doesn't work anymore. Here is the execution paused in `boot/home.js` right after runtime enters:
```javascript
module.exports = function(app, done) {
  const { About } = app.models;
  const router = app.loopback.Router();
  let challengeCount = 0;
  debugger
...
```
<img width="948" alt="image" src="https://user-images.githubusercontent.com/3822219/40638765-fda1de78-62c1-11e8-8aa4-bad324abb1c0.png">

and after I let it go with `continue (c)` command, the runtime throws an error while trying to access `entities` in the db:

<img width="946" alt="image" src="https://user-images.githubusercontent.com/3822219/40638842-7e5bb944-62c2-11e8-9527-15a21bca04a2.png">

According to `git blame` changes from #17150 could be related to the issue with seeding, since its code introduces call to `entities` while the server is being bootstrapped:

<img width="1210" alt="image" src="https://user-images.githubusercontent.com/3822219/40639167-4cbee9cc-62c4-11e8-9304-041e8e1bb49f.png">

**Comment by ptsurbeleu** [comment]
@Bouncey, can you take a look pls whether #17150 is the reason why seeding not working anymore?

**Comment by Bouncey** [comment]
Ohhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhh. Now I know what is broken, thank you for your investigation @ptsurbeleu!! 

The models are imported from the app during seed to destroy/create blocks and challenges. This has a side-effect of spinning up the server, which in its current state, requires challenge info whilst bootstrapping. 

I will see about patching this today

**Comment by ojongerius** [comment]
A hacky workaround provided by @raisedadead for the moment is:

Open [home.js](https://github.com/freeCodeCamp/freeCodeCamp/blob/staging/server/boot/home.js) and comment out line [12](https://github.com/freeCodeCamp/freeCodeCamp/blob/staging/server/boot/home.js#L12) until [24]( https://github.com/freeCodeCamp/freeCodeCamp/blob/staging/server/boot/home.js#L24 ). Run the seeding by either:

* Docker: `docker-compose run --rm freecodecamp npm run only-once`
* Other: `npm run only-once`

When seeding is done, restore [home.js](https://github.com/freeCodeCamp/freeCodeCamp/blob/staging/server/boot/home.js) by doing `git checkout server/boot/home.js` and start freeCodecamp as you normally would, ie:

* Docker: `docker-compose up`
* Other: `npm start`

Contributions for a permanent fix are welcome!

**Comment by rajatkantinandi** [comment]
@ojongerius 
Thanks.
Your solution works fine.
Just after saving the file & seeding the db we have to restore the file to original version otherwise the address localhost:3000 shows "Can't get /"
I think this step should be mentioned on contributing.md for now until there's a permanent fix.

**Comment by korydondzila** [comment]
I can work on this.

**Comment by ojongerius** [comment]
🎉

**Comment by Nirajn2311** [comment]
I'm sorry to say this, but it didn't work for me until I added stripe APIs to my .env file.

**Comment by ojongerius** [comment]
@Nirajn2311 that's a different issue though. The root cause of this issue has been fixed in #17472
